api/hasura/actions/_handlers/updateContribution.ts

```diff
@@ -0,0 +1,68 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { DateTime } from 'luxon';
+
+import { fetchAndVerifyContribution } from '../../../../api-lib/contributions';
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { errorResponseWithStatusCode } from '../../../../api-lib/HttpError';
+import { verifyHasuraRequestMiddleware } from '../../../../api-lib/validate';
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+  updateContributionInput,
+} from '../../../../src/lib/zod';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  const {
+    action: { name: actionName },
+    session_variables: { hasuraAddress: userAddress },
+    input: { payload },
+  } = composeHasuraActionRequestBodyWithSession(
+    updateContributionInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+
+  const { id, description, datetime_created } = payload;
+
+  const contribution = await fetchAndVerifyContribution({
+    id,
+    res,
+    userAddress,
+    operationName: actionName,
+  });
+
+  if (!contribution) return;
```

Comment:
seems to be missing an error response

Reply:
@zashton the error responses are inside the above function. If no contribution is returned, we cannot continue and must return.